### PR TITLE
updated assessment worker for AVS

### PR DESCRIPTION
**Pull request by vrindagoel-microsoft**

### CODE REVIEW

**Review by harshrathore-msft — CHANGES_REQUESTED** [review]

File: src/Assessment/Assess.cs [hunk 1]
```diff
@@ -204,10 +210,10 @@ public bool BeginAssessment()
                             SqlServicesVM.Add(discoverySiteMachine.MachineId);
                     }
 
-                    if (discoverySiteMachine.MachineId.Contains("vmwaresites"))
+                    if (discoverySiteMachine.MachineId.Contains("vmwaresites") || (UserInputObj.AzureMigrateSourceAppliances.Contains("import") && discoverySiteMachine.MachineId.Contains("importsites")))
                         AzureVMWareSolution.Add(assessmentSiteMachine);
 
-                    if (addMachineToGeneralVM)
+                    if (!UserInputObj.AzureMigrateSourceAppliances.Contains("import") && addMachineToGeneralVM)
```
Comment (harshrathore-msft) [inline src/Assessment/Assess.cs:216]: This is wrong,
We should only have "addMachineToGeneralVM" if it is not added to VMWARE, or any other source, it needs to be added to general VM,

make the addMachineToGeneralVm = false, in case it is import or gets added to vmware machines.

Should not augment this condition.

File: src/Azure-Migrate-Export.csproj [hunk 2]
```diff
@@ -157,8 +159,7 @@
     <Compile Include="Models\Assessment\Excel\CoreReport\SQL_All_Instances.cs" />
     <Compile Include="Models\Assessment\Excel\CoreReport\WebApp_PaaS.cs" />
     <Compile Include="Models\Assessment\Excel\CoreReport\Decommissioned_Machines.cs" />
-    <Compile Include="Models\Assessment\Excel\OpportunityReport\AVS_Summary.cs" />
-    <Compile Include="Models\Assessment\Excel\OpportunityReport\AVS_IaaS_Rehost_Perf.cs" />
+    <Compile Include="Models\Assessment\Excel\CoreReport\AVS_IaaS_Rehost_Perf.cs" />
```
Comment (harshrathore-msft) [inline src/Azure-Migrate-Export.csproj:162]: Why was this change required?

File: src/Excel/ExportCoreReport.cs [hunk 3]
```diff
@@ -399,9 +407,29 @@ private void Generate_All_VM_IaaS_Server_Rehost_Perf_Worksheet()
...
         }
 
+        private void Generate_AVS_Summary_Worksheet()
+        {
+            var dataWs = CoreWb.Worksheets.Add(CoreReportConstants.AVS_Summary_TabName, 16);
+
+            UtilityFunctions.AddColumnHeadersToWorksheet(dataWs, CoreReportConstants.AVS_Summary_Columns);
+
+            if (AVS_Summary_List != null && AVS_Summary_List.Count > 0)
+                dataWs.Cell(2, 1).InsertData(AVS_Summary_List);
```
Comment (harshrathore-msft) [inline src/Excel/ExportCoreReport.cs:417]: Need to add AVS AsOnPrem costs to AVS Summary Worksheet.

File: src/Excel/ExportCoreReport.cs [hunk 4]
```diff
@@ -399,9 +407,29 @@ private void Generate_All_VM_IaaS_Server_Rehost_Perf_Worksheet()
...
+
+        private void Generate_AVS_IaaS_Server_Rehost_Perf_Worksheet()
+        {
+            var dataWs = CoreWb.Worksheets.Add(CoreReportConstants.AVS_IaaS_Rehost_Perf_TabName, 17);
+
+            UtilityFunctions.AddColumnHeadersToWorksheet(dataWs, CoreReportConstants.AVS_IaaS_Rehost_Perf_Columns);
+
+            if (AVS_IaaS_Rehost_Perf_List != null && AVS_IaaS_Rehost_Perf_List.Count > 0)
+                dataWs.Cell(2, 1).InsertData(AVS_IaaS_Rehost_Perf_List);
+        }
```
Comment (harshrathore-msft) [inline src/Excel/ExportCoreReport.cs:428]: Need to add AVS_IaaS_Server_Rehost_AsOnPrem_Worksheet.

**Review by harshrathore-msft — CHANGES_REQUESTED** [review]

File: src/Azure-Migrate-Export.csproj (hunk 2 above)
Comment (vrindagoel-microsoft) [reply src/Azure-Migrate-Export.csproj:162]: I did this to shift all AVS excel worksheet related files to Core Report folder as mentioned in the previous comment: "Please ensure similar cases are handled everywhere.".

File: src/Assessment/Assess.cs (hunk 1 above)
Comment (vrindagoel-microsoft) [reply src/Assessment/Assess.cs:216]: without setting addMachineToGeneralVM as false, all the cases are working fine because of this condition:
!UserInputObj.AzureMigrateSourceAppliances.Contains("import")

File: src/Excel/ExportCoreReport.cs (hunk 3 above)
Comment (vrindagoel-microsoft) [reply src/Excel/ExportCoreReport.cs:417]: have already been handled

File: src/Excel/ExportCoreReport.cs [hunk 5]
```diff
@@ -399,9 +407,29 @@ private void Generate_All_VM_IaaS_Server_Rehost_Perf_Worksheet()
...
+
+        private void Generate_AVS_IaaS_Server_Rehost_Perf_Worksheet()
+        {
+            var dataWs = CoreWb.Worksheets.Add(OpportunityReportConstants.AVS_IaaS_Rehost_Perf_TabName, 17);
+
+            UtilityFunctions.AddColumnHeadersToWorksheet(dataWs, OpportunityReportConstants.AVS_IaaS_Rehost_Perf_Columns);
+
+            if (AVS_IaaS_Rehost_Perf_List != null && AVS_IaaS_Rehost_Perf_List.Count > 0)
+                dataWs.Cell(2, 1).InsertData(AVS_IaaS_Rehost_Perf_List);
+        }
```
[the remark this answers is not in the document]
Comment (vrindagoel-microsoft) [reply src/Excel/ExportCoreReport.cs]: done.

**Review by harshrathore-msft — APPROVED** [review]
